h2/src/main/org/h2/engine/Session.java

```diff
@@ -974,6 +976,8 @@ private void cleanTempTables(boolean closeSession) {
                         modificationId++;
                         table.setModified();
                         it.remove();
+                        // Exception thrown in org.h2.engine.Database.removeMeta if line below is missing with TestDeadlock
+                        database.lockMeta(this);
```

Comment:
Also commented out this "fix" line and got:
Exception in thread "main" org.h2.jdbc.JdbcSQLException: General error: "java.lang.RuntimeException: Unexpected code path" [50000-196]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:295)
	at org.h2.message.DbException.toSQLException(DbException.java:268)
	at org.h2.message.TraceObject.logAndConvert(TraceObject.java:358)
	at org.h2.jdbc.JdbcConnection.close(JdbcConnection.java:413)
	at org.h2.test.db.TestDeadlock.testTemporaryTablesAndMetaDataLocking(TestDeadlock.java:413)
	at org.h2.test.db.TestDeadlock.test(TestDeadlock.java:52)
	at org.h2.test.db.TestDeadlock.main(TestDeadlock.java:46)
Caused by: java.lang.RuntimeException: Unexpected code path
	at org.h2.message.DbException.throwInternalError(DbException.java:242)
	at org.h2.message.DbException.throwInternalError(DbException.java:255)
	at org.h2.engine.Database.removeMeta(Database.java:945)
	at org.h2.table.Table.removeChildrenAndResources(Table.java:553)
	at org.h2.table.TableView.removeChildrenAndResources(TableView.java:425)
	at org.h2.engine.Session.cleanTempTables(Session.java:981)
	at org.h2.engine.Session.close(Session.java:855)
	at org.h2.jdbc.JdbcConnection.close(JdbcConnection.java:405)
	... 3 more

Reply:
Latest travis build now reflects error I was seeing. Will revert these fixes back in, whne I have finished the ColumnName class... not long now...